app/src/main/java/edu/usf/sas/pal/muser/util/HeadphoneUtils.kt

```diff
@@ -0,0 +1,111 @@
+    /**
+     * Function to populate volume data class.
+     * @param context - context of the activity/fragment.
+     * @return VolumeData object.
+     */
+    @JvmStatic
+     fun getVolumeData(context: Context): VolumeData? {
+         val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager?
+         if (audioManager != null) {
+             val volumeLevel = audioManager.getStreamVolume(AudioManager.STREAM_MUSIC)
+             val volumeMax = audioManager.getStreamMaxVolume(AudioManager.STREAM_MUSIC)
+             var volumeMin = Int.MIN_VALUE
+             var volumeLevelDB = Float.MIN_VALUE
+             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                 volumeMin = audioManager.getStreamMinVolume(AudioManager.STREAM_MUSIC)
+                 volumeLevelDB = if (isSpeakerOn(context)) {
+                     audioManager.getStreamVolumeDb(AudioManager.STREAM_MUSIC,
+                             volumeLevel, AudioDeviceInfo.TYPE_BUILTIN_SPEAKER)
+                 } else {
+                     audioManager.getStreamVolumeDb(AudioManager.STREAM_MUSIC,
+                             volumeLevel, deviceType)
+                 }
+             }
+             return VolumeData(volumeLevel, volumeMax, volumeMin, volumeLevelDB)
+         }
+        return null
+     }
+
+    /**
+     * Function to check if the built in speaker is used or not
+     * @param context - context of the activity/fragment
+     * @return isSpeakerOn true or false.
+     */
+    @JvmStatic
+    fun isSpeakerOn(context: Context): Boolean{
+        val headsetTypes: Array<Int>
+        val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager?
+        var isSpeakerOn = true
+        if (audioManager != null) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                //  AudioDeviceInfo.TYPE_USB_HEADSET is only supported for devices with API_LEVEL > 26
+                headsetTypes = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    arrayOf(AudioDeviceInfo.TYPE_BLUETOOTH_A2DP, AudioDeviceInfo.TYPE_BLUETOOTH_SCO,
+                            AudioDeviceInfo.TYPE_WIRED_HEADSET, AudioDeviceInfo.TYPE_WIRED_HEADPHONES,
+                            AudioDeviceInfo.TYPE_USB_HEADSET, AudioDeviceInfo.TYPE_USB_DEVICE)
+                } else {
+                    arrayOf(AudioDeviceInfo.TYPE_BLUETOOTH_A2DP, AudioDeviceInfo.TYPE_BLUETOOTH_SCO,
+                            AudioDeviceInfo.TYPE_WIRED_HEADSET, AudioDeviceInfo.TYPE_WIRED_HEADPHONES,
+                            AudioDeviceInfo.TYPE_USB_DEVICE)
+                }
+                val audioDevices = audioManager.getDevices(AudioManager.GET_DEVICES_OUTPUTS)
+                if (audioDevices != null) {
+                    for (audioDevice in audioDevices) {
+                        Log.d(TAG, "isSpeakerOn: ${audioDevice.type}")
+                        if (headsetTypes.contains(audioDevice.type)) {
```

Comment:
I'm concerned that we're getting an array of `AudioDeviceInfo` but only saving the first `audioDevice` that appears in our `headsetTypes` list. It seems to me like we should be saving all of the `audioDevices`.